vncviewer/MonitorArrangement.cxx

```diff
@@ -0,0 +1,220 @@
+/* Copyright 2021 Hugo Lundin <huglu@cendio.se> for Cendio AB.
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <vector>
+#include <algorithm>
+
+#include <FL/Fl.H>
+#include <FL/Fl_Window.H>
+#include <FL/Fl_Box.H>
+#include <FL/Fl_Button.H>
+#include <FL/Fl_Check_Button.H>
+#include <FL/Fl_Timer.H>
+#include <FL/Fl_Group.H>
+#include <FL/fl_draw.H>
+#include <FL/Fl_PNG_Image.H>
+#include <FL/Fl_Check_Browser.H>
+
+#include "MonitorArrangement.h"
+
+#define MONITOR_ARRANGEMENT_MARGIN 20
+#define MONITOR_MARGIN_SCALE_FACTOR 0.99
+#define MONITOR_AVAILABLE_COLOR fl_lighter(fl_lighter(fl_lighter(FL_BACKGROUND_COLOR)))
+#define MONITOR_SELECTED_COLOR fl_rgb_color(53, 132, 228)
+#define MONITOR_REQUIRED_COLOR fl_lighter(fl_lighter(fl_rgb_color(53, 132, 228)))
+
+#define FL_CHECKERED_BOX (Fl_Boxtype) FL_FREE_BOXTYPE
+
+typedef struct {
+    unsigned int index;
+    MonitorArrangement *destination;
+} CallbackData;
+
+MonitorArrangement::MonitorArrangement(
+    int x, int y, int w, int h, MonitorArrangementDelegate *delegate
+): Fl_Group(x, y, w, h), m_delegate(delegate)
+{
+  box(FL_DOWN_BOX);
+  color(fl_lighter(FL_BACKGROUND_COLOR));
+
+  // Register a custom boxtype for the required monitor appearance.
+  Fl::set_boxtype(FL_CHECKERED_BOX, checkered_pattern_draw, 0, 0, 0, 0);
```

Comment:
Can this be called multiple times? I.e. if we would have multiple instances of this widget.

Reply:
`FL_CHECKERED_BOX` is an alias for `FL_FREE_BOXTYPE`  which is a pointer to the first free boxtype slot. Calling this function will set the callback for that type, so calling it repeatedly should not be a problem.   

However, another potential issue here is that if someone were to add another custom boxtype in the future it must be an alias to `FL_FREE_BOXTYPE+1`. Otherwise, this boxtype will break, when another callback is registered to the same free slot. I think that argues for that we might benefit from defining and registering all boxtypes in the same place.

Reply:
After discussing this with @CendioOssman, we decided to keep `FL_CHECKERED_BOX` defined here for now.